### DISCUSSION TITLE
Cache strategy in session.add

### DISCUSSION THREAD

**Discussion by kigawas** [sqlalchemy/sqlalchemy#6871] [Usage Questions] [answered]
It looks like the added instance will be cached and returned in the next `session.execute`:

```python
u = User(id=1, name='aaa')
session.add(u)
await session.commit()

u1 = (await session.execute(select(User).where(id=1))).scalar_one()
assert u is u1  #  True
```
I can work it around by `session.add(User(id=1, name='aaa'))`, but I'd like to know if there is any documentation on this.

**Comment by zzzeek** [comment, marked as answer]
yes, this is known as the identity map and is the fundamental paradigm the ORM works upon, background on this is at the following sources among others:

https://docs.sqlalchemy.org/en/14/glossary.html#term-identity-map

https://docs.sqlalchemy.org/en/14/orm/session_basics.html#what-does-the-session-do

https://docs.sqlalchemy.org/en/14/tutorial/orm_data_manipulation.html#identity-map

https://docs.sqlalchemy.org/en/14/orm/session_basics.html#is-the-session-a-cache
